collect_app/src/main/java/org/odk/collect/android/google/SheetsHelper.java

```diff
@@ -0,0 +1,146 @@
+    private Sheets sheets;
+
+    SheetsHelper(GoogleAccountCredential credential, HttpTransport transport, JsonFactory jsonFactory) {
+        // Initialize sheets service
+        sheets = new Sheets.Builder(transport, jsonFactory, credential)
+                .setApplicationName("ODK-Collect")
+                .build();
+    }
+
+    public void resizeSpreadSheet(String spreadsheetId, int sheetId, int columnSize) throws IOException {
+
+        // create grid properties with the new column size
+        GridProperties gridProperties = new GridProperties().setColumnCount(columnSize);
+
+        // create sheet properties for the first sheet in the spreadsheet
+        SheetProperties sheetProperties = new SheetProperties()
+                .setSheetId(sheetId)
+                .setGridProperties(gridProperties);
+
+        // Updates properties of the sheet with the specified sheetId
+        UpdateSheetPropertiesRequest updateSheetPropertyRequest = new UpdateSheetPropertiesRequest()
+                .setProperties(sheetProperties)
+                .setFields("gridProperties.columnCount");
+
+        // generate request
+        List<Request> requests = new ArrayList<>();
+        requests.add(new Request().setUpdateSheetProperties(updateSheetPropertyRequest));
+
+        // send the API request
+        sheets.spreadsheets()
+                .batchUpdate(
+                        spreadsheetId,
+                        new BatchUpdateSpreadsheetRequest().setRequests(requests)
+                ).execute();
+    }
+
+    /**
+     * Inserts a new row in the given sheet of the spreadsheet
+     */
+    public void insertRow(String spreadsheetId, String sheetName, ValueRange row) throws IOException {
+        sheets.spreadsheets().values()
+                .append(spreadsheetId, sheetName, row)
+                .setIncludeValuesInResponse(true)
+                .setValueInputOption("USER_ENTERED").execute();
+    }
+
+    /**
+     * Fetches the spreadsheet with the provided spreadsheetId
+     * <p>
+     * get(sheetId, range) method requires two parameters
+     * <p>
+     * since we want to search the whole sheet so we provide only the sheet name as range
+     * <p>
+     * range is in A1 notation
+     * eg. Sheet1!A1:G7
+     * <p>
+     * For more info   :   https://developers.google.com/sheets/api/reference/rest/
+     */
+    public List<List<Object>> getHeaderFeed(String spreadsheetId, String sheetName) throws IOException {
+        ValueRange response = sheets.spreadsheets()
+                .values()
+                .get(spreadsheetId, sheetName)
+                .execute();
+        return response.getValues();
+    }
+
+    /**
+     * Checks whether the selected google account has sufficient permissions
+     * to modify the given spreadsheetId. If yes, then returns complete spreadsheet
+     * otherwise throws exception
+     */
+    public Spreadsheet checkPermissions(String spreadsheetId) throws IOException {
```

Comment:
`updateSpreadsheetPermissions` might be a better name here, as its a bit confusing as to why we're returning a Spreadsheet from a method titled `check`.